### ISSUE TITLE
Integration of https://api.hackertarget.com/hostsearch

### ISSUE THREAD

**Issue by thomasmerz** [guelfoweb/knock#115]
Hi there,

do you know [hackertarget's hostsearch](https://api.hackertarget.com/hostsearch/?q=googlevideo.com)? All you need for "many" API calls is an API key.

Maybe you want or can integrate this, because this is not only a "dictionary attack", but a real "search "search for all subdomains associated with that domain": https://hackertarget.com/find-dns-host-records/

Kind regards!

**Comment by MC874** [comment]
> @guelfoweb , did you notice my issue and **how to improve your tool with real-life-data** and not only dictionary-guesses? 🤔

Knockpy work using a wordlist and online data from google, duckduckgo and virustotal queries. According to their ReadMe.

**Comment by thomasmerz** [comment]
One could add "using a wordlist and online data from … **and hackertarget's hostsearch API**" , if someone would like to add this real-data… 😉  I just wanted to raise an idea for a good improvement in my eyes.

**Comment by MC874** [comment]
> One could add "using a wordlist and online data from … **and hackertarget's hostsearch API**" , if someone would like to add this real-data… wink I just wanted to raise an idea for a good improvement in my eyes.

Yeah it's also good if they add another query source. There's spyse, dnsdb, dnsdumpster, censys, and passivetotal left to add. I honestly liked your idea 👌
